firmware/hw_layer/rtc_helper.cpp

```diff
@@ -4,23 +4,56 @@
  *
  * @date Feb 5, 2014
  * @author Andrey Belomutskiy, (c) 2012-2020
+ * @author Nathan Schulte, (c) 2022
  */
 
 #include "pch.h"
 
-#include <string.h> 
+#include <string.h>
 #include <time.h>
 
 #include "rtc_helper.h"
+
+#if EFI_PROD_CODE
 #include <sys/time.h>
 
+// Lua needs this function, but we don't necessarily have to implement it
+extern "C" int _gettimeofday(timeval* tv, void* tzvp) {
+	(void)tv; (void)tzvp;
+	return 0;
+}
+#endif // EFI_PROD_CODE
+
+#if EFI_RTC
+const char * ISO_UTC_DATE_TIME_FORMAT = "%04u-%02u-%02uT%02u:%02u:%02uZ";
+
+void convertRtcDateTimeToIsoUtc(RTCDateTime * timespec, char * isoUtcDateTime) {
+	uint32_t second = timespec->millisecond / 1000;
+	uint16_t minute = second / 60;
+	second -= minute * 60;
+	uint8_t hour = minute / 60;
+	minute -= hour * 60;
+	sprintf(isoUtcDateTime, ISO_UTC_DATE_TIME_FORMAT, timespec->year + RTC_BASE_YEAR, timespec->month, timespec->day, hour, minute, second);
+}
+
+void convertIsoUtcToRtcDateTime(char * isoDateTime, RTCDateTime * timespec) {
+	uint32_t year;
+	uint8_t month, day, hour, minute, second;
+	sscanf(ISO_UTC_DATE_TIME_FORMAT, isoDateTime, &year, &month, &day, &hour, &minute, &second);
+	timespec->year = year - RTC_BASE_YEAR;
+	timespec->month = month;
+	timespec->day = day;
+	timespec->millisecond = (((hour * 60) + minute) * 60 + second) * 1000;
```

Comment:
is this complex enough to justify a unit test or inclusion into any layer of testing?

Reply:
I'd say so.

I'm currently working toward 1s resolution; should I include millisecond/sub-second resolution instead?

Reply:
I would say discard ms, 1s is plenty for the relatively inaccurate clock on the ecu anyway.

Reply:
Where would I add these tests?  A [new] generic `rtc_helper` (or `hw_layer`?) module somewhere about `unit_tests`, or somehow else?  A stub module/file would go a long way to helping me implement this.